js/utils.js

```diff
@@ -73,12 +73,35 @@ function deleteIn(destObj, path) {
   return setIn(destObj, objPath, obj.without(propName));
 }
 
+function groupBy(array, predicate) {
```

Comment:
Related to issue #31. We can add a couple small utility functions that DRY up our code a lot.